### PR TITLE
Create wish list logic

### LINKED ISSUE #5227: Create Create Wish List feature
The new wish list effect will make use of the adapter created in the issue #5226. It will allow to make a cart "saved".

This issue will create:
- Wish list action with:
CREATE_WISH_LIST
CREATE_WISH_SUCCESS
CREATE_WISH_FAILED
&
LOAD_WISH_LIST
LOAD_WISH_SUCCESS
LOAD_WISH_FAILED
(not used yet)
- Wish list effect for create wish list
- Wish list reducer*
- Selector to get the wish list id

***Note** The wish list "state" is part of the multi-cart state and keeps the wish list code or uid (only code for now as you need to be authenticated). The wish list itself is saved among the list of carts.

### PR DESCRIPTION
**Pull request by LTiger14**
Closes #5227

### CODE REVIEW

File: projects/core/src/cart/store/reducers/wish-list.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { CartActions } from '../actions';
+
+export const initialState = '';
+
+export function wishListReducer(
```
Comment (WeizhengSap) [inline projects/core/src/cart/store/reducers/wish-list.reducer.ts]: How about move it into `multi-cart.reducer`?

Comment (WeizhengSap) [reply projects/core/src/cart/store/reducers/wish-list.reducer.ts]: When language/currency changed, does multi-cart data need to be reloaded?

File: projects/core/src/cart/store/reducers/multi-cart.reducer.ts [hunk 2]
```diff
@@ -42,6 +42,8 @@ export function cartEntitiesReducer(
   switch (action.type) {
     case CartActions.LOAD_MULTI_CART_SUCCESS:
     case CartActions.CREATE_MULTI_CART_SUCCESS:
+    case CartActions.CREATE_WISH_LIST_SUCCESS:
+    case CartActions.LOAD_WISH_LIST_SUCCESS:
       return action.payload.cart;
```
Comment (WeizhengSap) [inline projects/core/src/cart/store/reducers/multi-cart.reducer.ts]: what this reducer does?

Comment (LTiger14) [reply projects/core/src/cart/store/reducers/multi-cart.reducer.ts]: This reducer add the content of the wish list (cart) to the entities array that contains all the loaded carts

File: projects/core/src/cart/store/reducers/wish-list.reducer.ts (hunk 1 above)
Comment (LTiger14) [reply projects/core/src/cart/store/reducers/wish-list.reducer.ts]: I did think about moving it there will do. It should be reloaded I will make sure it's happening.

Comment (LTiger14) [reply projects/core/src/cart/store/reducers/wish-list.reducer.ts]: Moved, site context change will be handled in #5231

**Pull request merged**